src/include/log.h

```diff
@@ -84,8 +84,31 @@ extern "C" {
 #define	sys_printf(...)	syslog(LOG_NOTICE, __VA_ARGS__);
 #ifdef	DEBUG
 #define	DBPRT(x)	printf x;
-#else
-#define	DBPRT(x)
+#endif
+#ifdef DBPRT_LOG
+#include <stdlib.h>
+#include <errno.h>
+#include "libutil.h"
+#define STRIP_PARENS(...) __VA_ARGS__
+#undef DBPRT
+#define	DBPRT(x) \
+	if (will_log_event(PBSEVENT_DEBUG3)) { \
+		char * msg_; \
+		int msg_len_; \
+		int save_errno_ = errno; \
+		msg_len_ = pbs_asprintf(&msg_, STRIP_PARENS x); \
```

Comment:
Can you use log_eventf() here?  It's like log_event() and printf() combined into one.

Reply:
I used asprintf() to construct the message so that I could strip off the trailing newline, which was resulting in a considerable number of blank lines in the log files.  I might be able to add another macro layer to strip out the format string, but I would still need to copy it into another buffer in order to remove the newline.   While the asprintf() results in extra memory allocation, DBPRT only produces the extra code and memory overhead when DBPRT_LOG is defined, and I was only expecting it to be defined when `-O0 -g` had already been added and gdb wasn't helping.  In other words, when debugging frustration levels had reached critical.  :-)